apps/web/frontend/components/weekSelector.tsx

```diff
@@ -1,85 +1,69 @@
-import {
-  eachWeekOfInterval,
-  endOfWeek,
-  endOfYear,
-  format,
-  getWeek,
-  getYear,
-  isSameYear,
-  isThisWeek,
-  setWeek,
-  setYear,
-  startOfWeek,
-  startOfYear,
-} from 'date-fns'
+import { endOfWeek, format, getWeek, getYear, isThisWeek, startOfWeek, nextMonday, previousMonday } from 'date-fns'
+import { enGB } from 'date-fns/locale'
+import { useState } from 'react'
+import { BiLeftArrow, BiRightArrow } from 'react-icons/bi'
+import { BsCalendarCheck } from 'react-icons/bs'
+
+import { CalendarSelector } from './calendarSelector'
 
 interface WeekSelectorProps {
   value: Date
   onChange: (newDay: Date) => void
 }
 
 export const WeekSelector = ({ value, onChange }: WeekSelectorProps) => {
-  const selectedDay = startOfWeek(value, { weekStartsOn: 1 })
-  const selectedYear = getYear(selectedDay)
-  const selectedWeek = getWeek(selectedDay, { weekStartsOn: 1, firstWeekContainsDate: 7 })
-
-  const startOfSelectedYear = startOfYear(selectedDay)
-  const endOfSelectedYear = endOfYear(selectedDay)
-
-  const weeksOfSelectedYear = eachWeekOfInterval(
-    { start: startOfSelectedYear, end: endOfSelectedYear },
-    { weekStartsOn: 1 },
-  ).filter((startOfWeekDate) => isSameYear(startOfWeekDate, selectedDay))
+  const [selectedWeek, setSelectedWeek] = useState({
+    startDate: startOfWeek(value, { locale: enGB }),
+    endDate: endOfWeek(value, { locale: enGB }),
+    year: getYear(value),
+    key: 'selection',
+  })
```

Comment:
Please don't `useState()` here. You get the selected Day as a prop (`value`) from outside, and you can change the day with `onChange` prop. 

Also `key` seems useless and if you want to store 4 information you should use 4 `useStates`.